interfaces/sourcegen/src/sourcegen/csharp/generator.py

```diff
@@ -140,21 +144,21 @@ def _convert_func(self, parsed: CFunc) -> CsFunc:
             elif params:
                 params[0] = Param(handle_class_name, params[0].name)
 
-        for c_type, cs_type in self._config.ret_type_crosswalk.items():
-            if ret_type == c_type:
-                ret_type = cs_type
-                break
+        def crosswalk(par: str) -> str:
+            """Crosswalk of C/C# types."""
+            if par in self._config.c_type_crosswalk:
+                return self._config.c_type_crosswalk[par]
+            par = par.removeprefix("const ")  # C# doesn't recognize const
```

Comment:
I was confused why we need this const deletion for a scalar, but I see an example at `thermo_setDensity` where the parameter is marked const. As I follow up, I will probably add an extended comment explaining this.

Reply:
There are a couple of instances of `const double` in C++. The CLib header generator passes these through at this point. I think it would be best to remove the `const` in C++.